corporate/lib/support.py

```diff
@@ -260,26 +261,46 @@ def get_plan_data_for_support_view(
         current_plan=plan,
     )
 
-    if plan is not None:
-        new_plan, last_ledger_entry = billing_session.make_end_of_cycle_updates_if_needed(
-            plan, timezone_now()
+    if plan_data.current_plan is not None:
+        last_ledger_entry = (
+            LicenseLedger.objects.filter(
+                plan=plan_data.current_plan, event_time__lte=timezone_now()
+            )
+            .order_by("-id")
+            .first()
         )
-        if last_ledger_entry is not None:
-            if new_plan is not None:
-                plan_data.current_plan = new_plan  # nocoverage
+
+        if last_ledger_entry is None:  # nocoverage
+            # This shouldn't be possible because at least one
+            # license ledger entry should exist when a plan's
+            # status is less than CustomerPlan.LIVE_STATUS_THRESHOLD.
+            # But since we have a warning feature in the support
+            # view for plan data, we use that instead of raising
+            # an assertion error so that support staff can debug
+            # if this does ever occur.
+            plan_data.warning += "License ledger missing: No data for total licenses and revenue. "
+            plan_data.licenses = None
+            plan_data.annual_recurring_revenue = 0
+        else:
             plan_data.licenses = last_ledger_entry.licenses
-        assert plan_data.current_plan is not None  # for mypy
+            plan_data.annual_recurring_revenue = (
+                billing_session.get_annual_recurring_revenue_for_support_data(
+                    plan_data.current_plan, last_ledger_entry
+                )
+            )
 
         # If we already have user count data, we use that
         # instead of querying the database again to get
         # the number of currently used licenses.
         if stale_user_data:
-            plan_data.warning = USER_DATA_STALE_WARNING
+            plan_data.warning += USER_DATA_STALE_WARNING
+            plan_data.licenses_used = None
```

Comment:
This is None as the default, but I felt like it made this part of the function a bit more readable to see what is not being set when the audit log data is not updated.